cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/hvdc/CgmesDcConversion.java

```diff
@@ -58,41 +57,82 @@ public void convert() {
+            hvdc.add(nodeEquipment, islandEndHvdc1, islandEndHvdc2);
         });
 
         // Convert to IIDM each converter - dcLineSegment configuration
-        hvdc.getHvdcData().forEach(h -> convert(h.converters, h.dcLineSegments));
+        hvdc.getHvdcData().forEach(h -> convert(nodeEquipment, h.converters, h.dcLineSegments));
 
         // warnings
         context.dc().reportCgmesConvertersNotUsed();
         context.dc().reportCgmesDcLineSegmentNotUsed();
-
-        if (LOG.isDebugEnabled()) {
-            debugHvdc(adjacency, tpNodeEquipments, islands, islandsEnds, hvdc);
-        }
     }
 
     // Supported configurations are:
     //
+    // (1) One AcDcConverter pair One DcLineSegment
+    // (2) Two AcDcConverter pairs One DcLineSegment
+    // (3) One AcDcConverter pair Two DcLineSegments
+    //
+    private void convert(NodeEquipment nodeEquipment, List<HvdcConverter> converters, List<String> dcLineSegments) {
+        int converterNum = converters.size();
+        int dcLineSegmentNum = dcLineSegments.size();
+
+        if (converterNum == 1 && dcLineSegmentNum == 1) {
+            oneAcDcConverterPairOneDcLineSegment(nodeEquipment, converters, dcLineSegments);
+        } else if (converterNum == 2 && dcLineSegmentNum == 1) {
+            twoAcDcConverterPairsOneDcLineSegment(nodeEquipment, converters, dcLineSegments);
+        } else if (converterNum == 1 && dcLineSegmentNum == 2) {
+            oneAcDcConverterPairTwoDcLineSegments(nodeEquipment, converters, dcLineSegments);
+        } else {
+            throw new PowsyblException(String.format("Unexpected HVDC configuration: Converters %d DcLineSegments %d",
+                converterNum, dcLineSegmentNum));
+        }
+    }
+
+    // (1)
     //         CGMES Configuration                                                              IIDM configuration
     //
-    // (1)  AcDcConverterEnd1 --- DcLineSegment --- AcDcConverterEnd2        AcDcConverterEnd1 --- DcLineSegment --- AcDcConverterEnd2
+    //      AcDcConverterEnd1 --- DcLineSegment --- AcDcConverterEnd2        AcDcConverterEnd1 --- DcLineSegment --- AcDcConverterEnd2
+    //
+    private void oneAcDcConverterPairOneDcLineSegment(NodeEquipment nodeEquipment, List<HvdcConverter> converters,
+        List<String> dcLineSegments) {
+        if (isHvdcWellOriented(nodeEquipment, converters.get(0).acDcConvertersEnd1, dcLineSegments.get(0))) {
+            convert(converters.get(0).acDcConvertersEnd1, converters.get(0).acDcConvertersEnd2, dcLineSegments.get(0));
+        } else {
+            convert(converters.get(0).acDcConvertersEnd2, converters.get(0).acDcConvertersEnd1, dcLineSegments.get(0));
+        }
+    }
+
+    // (2)
+    //         CGMES Configuration                                                              IIDM configuration
     //
-    // (2)  AcDcConverter1End1 ---                 --- AcDcConverter1End2
+    //      AcDcConverter1End1 ---                 --- AcDcConverter1End2
```

Comment:
Thanks a lot for the comments, I think we should put these clear explanations in our web site documentation for CGMES DC lines conversion.